CRM/Event/Form/Participant.php

```diff
@@ -2025,9 +2025,13 @@ protected function sendReceipts($params, array $participants, $lineItem, $additi
     $this->assign('customGroup', $customGroup);
 
     $fromEmails = CRM_Event_BAO_Event::getFromEmailIds($this->getEventID());
-    foreach ($this->_contactIds as $num => $contactID) {
-      [, $email, $doNotEmail] = CRM_Contact_BAO_Contact::getContactDetails($contactID);
-      if (!$email || $doNotEmail) {
+    foreach ($participants as $num => $participant) {
+      $participantID = $participant->id;
+      $contactID = $participant->contact_id;
+      $key = 'contact_' . $contactID;
+
+      $this->define('Contact', $key, ['id' => $contactID]);
```

Comment:
This definitely wasn't my vision for how `$key` would allow you to lookup a contact without needing to know the id! But ok...

Reply:
@colemanw yeah the use of the lookup in this function is marginal tbh - it was already there so I left it in